client/maisie/resources/models.py

```diff
@@ -93,17 +96,27 @@ def download(self, id: int):
             request = request.json()
             if (
                 ("data") in request
+                and "checksum" in request["data"]
                 and "_links" in request["data"]
                 and "name" in request["data"]
                 and "download" in request["data"]["_links"]
             ):
                 download_link = request["data"]["_links"]["download"]
                 download_data = session.get(download_link)
                 model_name = request["data"]["name"]
-        if path:
+                download_checksum = request["data"]["checksum"]
```

Comment:
`download_checksum` does not seem to be a good fit for the name of this variable.
It is neither the checksum of an already *downloaded* file, nor the checksum to *be downloaded* (?). 
This name would probably be more fitting for a method, not a variable name.

What about some of these: `source_checksum`, `original_checksum`, `stored_checksum` (as opposed to `calculated_checksum`/`local_checksum`) or maybe even `expected_checksum`?